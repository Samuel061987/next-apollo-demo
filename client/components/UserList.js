import React from 'react';
import styles from "../styles/Users.module.css";
const UserList = ({user}) => {
    return(
        <div className={styles.grid}>
        {user && user.map((data,index) => (
            <div key={index} className={styles.card}>
              <h3>{data.name}</h3>
              <p data-testid="phone">
              Phone: {data.phone}
              </p>
              <p data-testid="address">
              Address: {data.address}
              </p>
              <p data-testid="email">
              Email: {data.email}
              </p>
            </div>
          ))}
         </div>
    )
}
export default UserList;