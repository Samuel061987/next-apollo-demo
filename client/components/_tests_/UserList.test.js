import React from 'react';
import {render,cleanup} from '@testing-library/react';
import '@testing-library/jest-dom';
import UserList from '../UserList';

afterEach(()=>{
    cleanup();
});

const renderComponent=(user='')=>{
    return(<UserList user={user}></UserList>)
}

describe('unit testing for the userList Component',()=>{
    describe('visual snapshot testing',()=>{
        it('render userlist component',()=>{
            const mockData=[
                { name: "sam", address: "add", phone: "012345", email: "sam@gmail" },
            ];
            const {container}= render(renderComponent(mockData));
            expect(container).toMatchSnapshot();
        });
        it('render userlist component and check the element rendered',()=>{
            const mockData=[
                { name: "sam", address: "add", phone: "012345", email: "sam@gmail" },
            ];
            const {getByTestId}= render(renderComponent(mockData));
            expect(getByTestId('phone')).toBeInTheDocument();
            expect(getByTestId('address')).toBeInTheDocument();
            expect(getByTestId('email')).toBeInTheDocument();
        });

        it('render userlist component without data',()=>{
            const {queryByTestId}= render(renderComponent());
            expect(queryByTestId('phone')).not.toBeInTheDocument();
            expect(queryByTestId('address')).not.toBeInTheDocument();
            expect(queryByTestId('email')).not.toBeInTheDocument();
        });
    });

})