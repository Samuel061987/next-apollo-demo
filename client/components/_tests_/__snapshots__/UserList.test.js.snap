// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit testing for the userList Component visual snapshot testing render userlist component 1`] = `
<div>
  <div
    class="grid"
  >
    <div
      class="card"
    >
      <h3>
        sam
      </h3>
      <p
        data-testid="phone"
      >
        Phone: 
        012345
      </p>
      <p
        data-testid="address"
      >
        Address: 
        add
      </p>
      <p
        data-testid="email"
      >
        Email: 
        sam@gmail
      </p>
    </div>
  </div>
</div>
`;
